src/query/service/src/sql/planner/binder/ddl/table.rs

```diff
@@ -757,10 +761,22 @@ impl<'a> Binder {
                     let name = normalize_identifier(&column.name, &self.name_resolution_ctx).name;
                     let data_type = TypeFactory::instance().get(column.data_type.to_string())?;
 
-                    let field = DataField::new(&name, data_type).with_default_expr({
+                    let field = DataField::new(&name, data_type.clone()).with_default_expr({
                         if let Some(default_expr) = &column.default_expr {
-                            scalar_binder.bind(default_expr).await?;
-                            Some(default_expr.to_string())
+                            let (mut expr, expr_type) = scalar_binder.bind(default_expr).await?;
+                            if compare_coercion(&data_type, &expr_type).is_err() {
+                                return Err(ErrorCode::SemanticError(format!("column {name} is of type {} but default expression is of type {}", data_type, expr_type)));
+                            }
+                            if !expr_type.eq(&data_type) {
+                                expr = Scalar::CastExpr(CastExpr {
+                                    argument: Box::new(expr),
+                                    from_type: Box::new(expr_type),
+                                    target_type: Box::new(data_type),
+                                })
+                            }
+                            let mut builder = PhysicalScalarBuilder;
+                            let serializable_expr = builder.build(&expr)?;
+                            Some(serde_json::to_string(&serializable_expr)?)
```

Comment:
We should serialize the default expression into SQL string to prevent compatibility issues.